### PR TITLE
FEAT: Create shader system

### PR DESCRIPTION
**Pull request by Peralban**
Create shader initialisation and display.
At the beginning, we need a Settings.json file in Game/Config
image

The whole shader will be saved in a map, and if "current" changes, the shader will be applied with the function I created.

Example of shader utilisation:

image
image

---

image
image

---

image
image

---

image
image

### CODE REVIEW

File: Game/Assets/player.gif
Comment (Thomaltarix) [inline Game/Assets/player.gif:1]: The file is already existing as `Game/Assets/players.gif`

File: Game/Config/Settings.json
Comment (Thomaltarix) [inline Game/Config/Settings.json:1]: I think you should rename the file `ShaderSettings.json` or putting it into a sub-folder Shaders in the Config one

File: GraphicLibrary/Src/GraphicLib.hpp [hunk 1]
```diff
@@ -137,8 +205,12 @@ class GraphicLib : public IGraphic {
 
     private:
 
+        bool _isShaderReady();
```
Comment (Thomaltarix) [inline GraphicLibrary/Src/GraphicLib.hpp:208]: A little comment would be nice 👍

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
Some little changes about comments or file names

File: Engine/Client/Src/InitShader/InitShader.hpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * In our game one example of shader is the system to adapt color for the colorblind.
+ */
+class InitShader {
+
+public:
+
+    class JsonError : public AError {
+        public:
+            JsonError(const std::string &message) : AError(message) {}
+    };
```
Comment (mathieurobert1) [inline Engine/Client/Src/InitShader/InitShader.hpp:31]: Add comments to this

File: Game/Assets/player.gif
Comment (mathieurobert1) [reply Game/Assets/player.gif:1]: Yep

File: Game/Config/Settings.json
Comment (Peralban) [reply Game/Config/Settings.json:1]: No, in Settings, there will be a multitude of settings, including quality settings. Therefore, we do not need to divide all our JSON files, as we can combine all the settings into a single file.

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]

**Pull request merged**